dhcpv4/client.go

```diff
@@ -267,28 +262,30 @@ func (c *Client) sendReceive(sendFd, recvFd int, packet *DHCPv4, messageType Mes
 	// Create a goroutine to perform the blocking send, and time it out after
 	// a certain amount of time.
 	var (
-		destination [4]byte
+		destination [net.IPv4len]byte
 		response    *DHCPv4
 	)
 	copy(destination[:], raddr.IP.To4())
 	remoteAddr := unix.SockaddrInet4{Port: laddr.Port, Addr: destination}
 	recvErrors := make(chan error, 1)
 	go func(errs chan<- error) {
-		conn, innerErr := net.FileConn(os.NewFile(uintptr(recvFd), ""))
-		if innerErr != nil {
+		// set read timeout
+		timeout := unix.Timeval{
+			Sec:  int64(c.ReadTimeout.Seconds()),
+			Usec: 0,
+		}
+		if innerErr := unix.SetsockoptTimeval(recvFd, unix.SOL_SOCKET, unix.SO_RCVTIMEO, &timeout); innerErr != nil {
 			errs <- innerErr
 			return
 		}
-		defer conn.Close()
-		conn.SetReadDeadline(time.Now().Add(c.ReadTimeout))
-
 		for {
 			buf := make([]byte, MaxUDPReceivedPacketSize)
-			n, _, _, _, innerErr := conn.(*net.UDPConn).ReadMsgUDP(buf, []byte{})
+			n, innerErr := unix.Read(recvFd, buf)
 			if innerErr != nil {
 				errs <- innerErr
 				return
 			}
+			log.Print("SOMETHING")
```

Comment:
I guess this is going away?

Reply:
yes, the last commit was more "saving the code" to iterate, it doesn't even pass tests